src/validate.ts

```diff
@@ -27,128 +27,186 @@ import {
+      `${badCommits.length} of the pull request's commits are not valid Conventional Commits`
     );
+    for (const c of badCommits) {
+      outputErrors(c.message, c.errors, c.sha);
+    }
+  }
+
+  return {
+    compliant: badCommits.length === 0,
+    messages: conventionalCommitMessages,
+  };
+}
+
+/**
+ * Validates the pull request title and, if compliant, returns it as a
+ * ConventionalCommitMessage object.
+ */
+export async function validatePrTitle(
+  config: Configuration
+): Promise<ConventionalCommitMessage | undefined> {
+  const prTitleText = await getPullRequestTitle();
+  let errors: LlvmError[] = [];
+  let conventionalCommitMessage: ConventionalCommitMessage | undefined;
+
+  core.info(""); // for vertical whitespace
+  const baseError =
+    "The pull request title is not compliant with the Conventional Commits specification";
+  try {
+    conventionalCommitMessage = new ConventionalCommitMessage(prTitleText);
+  } catch (error) {
+    if (error instanceof ConventionalCommitError) {
+      errors = error.errors;
+    } else if (error instanceof MergeCommitError) {
+      throw new Error(`${baseError}; describes a merge commit`);
+    } else if (error instanceof FixupCommitError) {
+      throw new Error(`${baseError}; describes a fixup commit`);
+    } else {
+      throw new Error(baseError);
+    }
+  }
+  if (errors.length > 0) {
+    core.setFailed(baseError);
+    outputErrors(prTitleText, errors, undefined);
   } else {
-    core.info(
-      "✅ Your Pull Request complies with the Conventional Commits specification!"
+    core.startGroup(
+      `✅ The pull request title is compliant with the Conventional Commits specification`
     );
+    core.info(prTitleText);
+    core.endGroup();
   }
-
-  return conventionalCommitMessages;
+  return conventionalCommitMessage;
 }
 
 /**
- * Validates bump level consistency between the PR title and its commits
+ * Validates bump level consistency between the PR title and its commits.
+ * This implies that the PR title must comply with the Conventional Commits spec.
  */
 export async function validatePrTitleBump(
   config: Configuration
-): Promise<void> {
+): Promise<boolean> {
   const prTitleText = await getPullRequestTitle();
   const commits: string[] = (await getCommits(getPullRequestId())).map(m => {
     return m.commit.message;
   });
   let highestBump: SemVerType = SemVerType.NONE;
   let atLeastOneConventionalCommitFound = false;
-  const prTitle = (() => {
-    try {
-      return new ConventionalCommitMessage(prTitleText);
-    } catch (error) {
-      throw new Error(
-        `The PR title does not conform to the Conventional Commits specification.`
-      );
-    }
-  })();
+  let nonCompliantCommitCount = 0;
+  const prTitle = await validatePrTitle(config);
+
+  if (prTitle === undefined) {
+    core.warning(
+      "Cannot validate bump level, as PR title is not a valid " +
+        "Conventional Commit message"
+    );
+    return true;
```

Comment:
Shouldn't this be returning `false` instead - to handle the errors found during `validatePrTitle`?

Reference usage;
```
      const ok = await validatePrTitleBump(config);
      compliant &&= ok;
```
snippppp
```
    if (compliant) {
       core.info("✅ The pull request passed all configured checks");
     }
```

Reply:
Yep, good catch, this is left-over "intermediate" logic.